shelf/bulk_update/utils.py

```diff
@@ -16,6 +15,40 @@ def run(args):
                 options provided to the update-search-index
                 script
     """
+    logger = set_up_logger("update-search-index", args)
+    config = get_config(args, logger.level)
+    bucket_list = get_bucket_list(args)
+    container = Container(config, logger)
+    runner = container.create_runner(update_search_index)
+    runner.run(bucket_list)
+
+
+def run_search_prune(args):
```

Comment:
You should have a generic run function which takes two parameters. 1.) the function you wish run in the subprocess and 2.) the name of the logger.